source/common/http/http1/codec_impl.cc

```diff
@@ -1081,6 +1081,12 @@ Status ServerConnectionImpl::sendProtocolError(absl::string_view details) {
     sendProtocolErrorOld(details);
     return okStatus();
   }
+
+  // Cannot send a message using the HCM if the stream was reset.
+  if (resetStreamCalled()) {
+    return codecClientError("sendProtocolError on a reset stream should not be called");
+  }
```

Comment:
Per offline conversation, personally I don't think it makes sense to add error handling here for something that should not happen. I think this is just a fuzzer bug. If we want to add actual code to the codec, I would choose between ASSERT/RELEASE_ASSERT/ENVOY_BUG, but I think we should have one of those here and below (with tests).

Reply:
Thanks.
I've reverted the changes to the codec files.
As you've said, it should not happen in practice, and in addition the code that fetches the value (from absl::optional) will throw an exception in cases where the assertion fails.